local-interchain/interchain/start.go

```diff
@@ -137,6 +144,40 @@ func StartChain(installDir, chainCfgFile string, ac *AppConfig) {
 		LinkIBCPaths(ibcpaths, chains, ic, relayer)
 	}
 
+	// Add Interchain Security chain pairs together
+
+	if len(icsPair) > 0 {
+
+		// iterate icsPair
+		for provider, consumers := range icsPair {
+			var p ibc.Chain
+			var c ibc.Chain
+
+			// a provider can have multiple consumers
+			for _, consumer := range consumers {
+				for _, chain := range chains {
+					if chain.Config().ChainID == provider {
+						p = chain
+					}
+					if chain.Config().ChainID == consumer {
+						c = chain
+					}
+				}
+			}
+
+			pathName := fmt.Sprintf("%s-%s", p.Config().ChainID, c.Config().ChainID)
+
+			logger.Info("Adding ICS pair", zap.String("provider", p.Config().ChainID), zap.String("consumer", c.Config().ChainID), zap.String("path", pathName))
+
+			ic = ic.AddProviderConsumerLink(interchaintest.ProviderConsumerLink{
+				Provider: p,
+				Consumer: c,
+				Relayer:  relayer,
+				Path:     pathName,
+			})
+		}
+	}
```

Comment:
this could be its own function, which would increase readability and make it easier to maintain.

Reply:
atm many places in the `StartChain` could use this. It's kinda a monolith for starting up. Need to do a refactor here in the near future